agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -4319,6 +4328,154 @@ bool backhaul_manager::add_link_metrics(const sMacAddr &reporter_al_mac,
+
+    bool ret = false;
+
+    std::vector<std::tuple<sMacAddr, int8_t>> transmit_power;
+
+    for (auto &transmit_power_limit_tlv : cmdu_rx.getClassList<wfa_map::tlvTransmitPowerLimit>()) {
+        if (!transmit_power_limit_tlv) {
+            LOG(ERROR) << "Failed getClassList for tlvTransmitPowerLimit";
+            continue;
+        }
+        transmit_power.push_back(
+            std::tuple<sMacAddr, int8_t>(transmit_power_limit_tlv->radio_uid(),
+                                         transmit_power_limit_tlv->transmit_power_limit_dbm()));
+    }
+
+    for (auto &channel_selection : cmdu_rx.getClassList<wfa_map::tlvChannelPreference>()) {
+        for (auto &socket : slaves_sockets) {
+            auto mac = channel_selection->radio_uid();
+
+            if (socket->radio_mac != mac) {
+                continue;
+            }
+            auto forward =
+                cmdu_tx.create(mid, ieee1905_1::eMessageType::CHANNEL_SELECTION_REQUEST_MESSAGE);
+            if (!forward) {
+                LOG(ERROR) << "Failed to create CHANNEL_SELECTION_REQUEST_MESSAGE";
+                ret = false;
+            }
+            //TODO: find the way for copy tlvChannelPreference from rx to tx
+            auto channel_preference = cmdu_tx.addClass<wfa_map::tlvChannelPreference>();
+            if (!channel_preference) {
+                LOG(ERROR) << "Failed addClass<wfa_map::tlvChannelPreference>";
+                ret = false;
+            }
+            channel_preference->radio_uid() = mac;
+
+            auto op_class_channels_tx = channel_preference->create_operating_classes_list();
+            if (!op_class_channels_tx) {
+                LOG(ERROR) << "create_operating_classes_list() has failed!";
+                return false;
+            }
+
+            if (!op_class_channels_tx->alloc_channel_list()) {
+                LOG(ERROR) << "alloc_channel_list() has failed!";
+                return false;
+            }
+
+            auto power_limit_tlv = cmdu_tx.addClass<wfa_map::tlvTransmitPowerLimit>();
+            if (!power_limit_tlv) {
+                LOG(ERROR) << "Failed addClass<wfa_map::tlvTransmitPowerLimit>";
+                ret = false;
+            }
+
+            power_limit_tlv->radio_uid() = mac;
+
+            for (auto &power : transmit_power) {
+                if (std::get<0>(power) != mac) {
+                    continue;
+                }
+                power_limit_tlv->transmit_power_limit_dbm() = std::get<1>(power);
+            }
+
+            //if (!message_com::send_cmdu(socket->slave, cmdu_tx)) {
+            //    LOG(ERROR)
+            //        << "Failed forwarding CHANNEL_SELECTION_REQUEST_MESSAGE message to son_slave";
+            //    ret = false;
+            //    continue;
+            //}
+
+            m_channel_selection_request.push_back(mac);
+            ret = true;
+            break;
+        }
+    }
```

Comment:
All this is not needed, because we expect a response from *all* slaves. So what we need to do is simply to populate m_channel_selection_request with all slaves. And it has to be forwarded to the slaves, but for that we can simply return false.

Reply:
Fixed